### PR TITLE
Mat 6793 transferred qdm measure cql error 2

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-6793](https://jira.cms.gov/browse/MAT-6793)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/cms/gov/madie/measure/services/ElmTranslatorClient.java [hunk 1]
```diff
@@ -45,7 +47,21 @@ public boolean hasErrors(ElmJson elmJson) {
...
       ObjectMapper mapper = new ObjectMapper();
       JsonNode jsonNode = mapper.readTree(elmJson.getJson());
-      return jsonNode.has("errorExceptions") && jsonNode.get("errorExceptions").size() > 0;
+      boolean parsingErrors = false;
+      if (jsonNode.has("errorExceptions")) {
+        JsonNode errorExceptionsNode = jsonNode.get("errorExceptions");
+        Iterator<JsonNode> entyIter = errorExceptionsNode.iterator();
+        while (entyIter.hasNext()) {
+          JsonNode errorNode = entyIter.next();
+          if (errorNode.get("type") != null
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/services/ElmTranslatorClient.java:56]: we are only checking for parsing errors. what are the other types of errors that we want to ignore?

Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/services/ElmTranslatorClient.java:56]: @adongare Only parsing errors and externalErrors, to be consistent with frontend.

Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/services/ElmTranslatorClient.java:56]: @adongare here are some example errors from errorExceptions: 401 : [no body], Could not resolve identifier AdultOutpatientEncounters in the current library. Member Qualifying Encounters not found for type null. Could not determine signature for invocation of operator System.Exists....I think some are syntax errors some are errors from global library?? Anyway, these errors are ignored by frontend, frontend only deals externalErrors, and parsing errors as well.

Comment (adongare) [reply src/main/java/cms/gov/madie/measure/services/ElmTranslatorClient.java:56]: Go it, thanks for the explanation.

Comment (adongare) [reply src/main/java/cms/gov/madie/measure/services/ElmTranslatorClient.java:56]: @jkotanchik-SB should we not consider all the errors except warnings? syntax errors or errros from included library makes the CQL invalid as well and they need to know, right?

Comment (jkotanchik-SB) [reply src/main/java/cms/gov/madie/measure/services/ElmTranslatorClient.java:56]: At this point in the import flow it looks like the CQL is translated by MADiE using the Libraries in MADiE.

If so, then there _shouldn't_ be any errors in the included libraries as they must be versioned to be used in translation. "Shouldn't" meaning there is an edge case here if the translator is feed a non-versioned Library.

Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/services/ElmTranslatorClient.java:56]: @jkotanchik-SB @adongare It looks what the transfer measure is doing is the same as creating a measure, it does get the elmJson from translator, and then check if there is error. I do see errorExceptions but it looks like they are ignored by the frontend editor, only externalErrors are considered an error.

**Review by adongare — APPROVED** [review]

**Pull request merged**